samcli/lib/sync/infra_sync_executor.py

```diff
@@ -502,6 +528,25 @@ def get_template(self, template_path: str) -> Optional[Dict]:
 
         return template
 
+    def _get_stack_parameters(self, stack_name: str) -> List[Dict[str, str]]:
+        """
+        Returns the stack parameters for a given stack
+
+        Parameters
+        ----------
+        stack_name: str
+            The name of the stack
+
+        Returns
+        -------
+            List of Dicts in the form { 'ParameterKey': Foo, 'ParameterValue': Bar }
+
+        """
+        return cast(
+            List[Dict[str, str]],
+            self._cfn_client.describe_stacks(StackName=stack_name).get("Stacks")[0].get("Parameters", []),
```

Comment:
This might fail if stack haven't been deployed yet. For that reason I would rather check the size of the collection first instead of accessing the `[0]` element there.

Reply:
It's possible for the `describe_stacks` call to succeed for a given stack name but not have any elements in the `Stacks`? 

interesting, will do